vike/client/client-routing-runtime/getPageContextFromHooks.ts

```diff
@@ -165,8 +170,9 @@ async function getPageContextAlreadyRouted(
     }
   }
 
-  {
+  if (!pageContextFetchedFromServer) {
```

Comment:
this is the fix. All the rest of the PR is just comments and typescript-happiness